opengever/maintenance/scripts/rename_deployment.py

```diff
@@ -150,7 +154,35 @@ def reindex_objects(plone, options):
                 # full text searches. Only IBaseDocument has an indexer for it.
                 attrs_to_index.append('metadata')
 
+        if options.new_au_title and options.migrate_filing_numbers:
+            # Filing numbers are dependent on AU title:
+            # Rewrite them using using string replacement
+
+            if IFilingNumberMarker.providedBy(obj):
+                old_fn = IFilingNumber(obj).filing_no
+
+                if old_fn and old_fn.startswith(old_au_title):
+                    new_au_title = options.new_au_title.decode('utf-8')
+                    new_fn = old_fn.replace(old_au_title, new_au_title, 1)
+
+                    logger.info('Migrating filing number %r to %r for %r' % (
+                        old_fn, new_fn, obj))
+
+                    if not options.dryrun:
+                        IFilingNumber(obj).filing_no = new_fn
+
+                        # December 2020. Not taking any chances.
+                        IAnnotations(obj)['filing_no_before_rename'] = old_fn
+
+                # Reindex all objects with FN behavior (dossiers), even if
+                # no FN was found and changed. They might already have been
+                # issued a prefix, but not be closed yet. Then they have a
+                # ...-Amt-? number which still needs to be reindexed.
+                if not options.dryrun:
+                    attrs_to_index.extend(['filing_no', 'searchable_filing_no', 'SearchableText'])
+
         if attrs_to_index:
+            attrs_to_index = list(set(attrs_to_index))
```

Comment:
Is there a chance that we end up with a field that is solr only in that list, for example `metadata`?
If so I'd propose to add `UID` here to make sure we are not reindexing the whole object...

Reply:
Good point, thank you. Updated.